aws-datastore/src/main/java/com/amplifyframework/datastore/syncengine/MutationProcessor.java

```diff
@@ -47,24 +57,40 @@
     private static final Logger LOG = Amplify.Logging.forNamespace("amplify:aws-datastore");
     private static final long ITEM_PROCESSING_TIMEOUT_MS = TimeUnit.SECONDS.toMillis(10);
 
-    private final VersionRepository versionRepository;
     private final Merger merger;
-    private final AppSync appSync;
+    private final VersionRepository versionRepository;
+    private final SyncTimeRegistry syncTimeRegistry;
     private final MutationOutbox mutationOutbox;
+    private final AppSync appSync;
+    private final DataStoreConfigurationProvider configurationProvider;
     private final CompositeDisposable ongoingOperationsDisposable;
 
-    MutationProcessor(
+    private MutationProcessor(
```

Comment:
Can you change this to take a `Builder` instead of a bunch of parameters?

Reply:
Done!